test/driver.cpp

```diff
@@ -56,18 +59,71 @@ int main(int argc, char* argv[])
 
 
 	try {
-		runTests();
+		run_suites(options);
 	}
-	catch (const std::string& msg) {
-		logLine(msg.data());
-		if (options.fom == file_open_mode::no_file)
-			show_error(msg.data());
+	catch (const cmd_line_error& cle) {
+		show_error_and_usage(cle.what(), argv[0]);
+		return -5;
+	}
+	catch (const std::exception& e) {
+		show_error((std::string{ "Unexpected error: " } +e.what()).data());
+		return -6;
+	}
+	catch (...) {
+		show_error("Unexpected error");
+		return -7;
 	}
 
 	if (options.summary)
-		summarizeTests();
+		summarize_tests();
 
-	return getTestsFailed();
+	return get_tests_failed_total();
+}
+
+
+//declare suite runners: one for each test suite
+//also add an entry in the "suites" map for each suite defined
+void array_test();
+
+
+void run_suites(const Options& options)
+{
+	using suite_runner_type = void(*)();
+	std::map<std::string, suite_runner_type> suites{
+		//add one entry for each test suite to run
+		//suite-name and suite-runner, both in braces
+		{"array-test", array_test}
+	};
+
+	//build a collection of suites to run if necessary
+	//options.suites_to_run is empty or a semi-colon delimited list of suite names
+	const auto suites_to_run = split(options.suites_to_run, ';');
+	auto run_all_suites = suites_to_run.empty();
+
+	//check that the suites specified are actually defined
+	//this check is not required to run the suites, but is included to inform the user of the issue
+	//silently ignoring leaves the user unaware of the reason a specified suite doesn't run
+	if (!run_all_suites) {
+		auto end_suites = suites.cend();
+		for (auto& suite_name : suites_to_run) {
+			if (suites.find(suite_name) == end_suites) {
+				assert(false);
+				throw invalid_option_value{ std::string{"suite "} +suite_name + " not defined" };
```

Comment:
```suggestion
				throw invalid_option_value{ std::string{"suite "} + suite_name + " not defined" };
```

Reply:
I'm glad you noticed this issue too. However, at least in my environment, formatting the document oddly remove that space between `+` and `suite_name`. Any idea what might be going on?